File: 201_BitwiseANDofNumbersRange.py
##82.81%
class Solution(object):
    def rangeBitwiseAnd(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        if m == n or m == 0:
            return m
        cnt = 0
        while m != 0 and m != n:
            m >>= 1
            n >>= 1
            cnt += 1
        if m == 0:
            return 0
        else:
            return m << cnt

File: 201_BitwiseANDofNumbersRange.cpp
//80.06%
class Solution {
public:
    int rangeBitwiseAnd(int m, int n) {
        if(m == n || m == 0)
			return m;
		int cnt(0);
		while(m && m != n){
			m >>= 1;
			n >>= 1;
			++cnt;
		}
		if(!m)
			return 0;
		else
			return m << cnt;
    }
};

int rangeBitwiseAnd(int m, int n) {
    return (n > m) ? (rangeBitwiseAnd(m/2, n/2) << 1) : m;
}

//A java solution---which is really the best
 public int rangeBitwiseAnd(int m, int n) {
        while(m<n) n = n & (n-1);
        return n;
}